Sources/Storage/Storage.swift

```diff
@@ -10,13 +10,13 @@ import Foundation
 
 protocol Storage {
     associatedtype Key
```

Comment:
Hmm, thinking about it, we do need to `Key` to be `Hashable`. Add the constraint for it to the protocol